core/src/fees.ts

```diff
@@ -23,16 +26,42 @@ export const getApproximateTransactionFees = async function (network: string): P
     };
 };
 
-export const enrichAuctionWithTransactionFees = function (
+export const enrichAuctionWithTransactionFees = async function (
     auction: Auction,
-    fees: TransactionFees
-): AuctionTransaction {
+    fees: TransactionFees,
+    network: string
+): Promise<AuctionTransaction> {
+    let totalFeeETH = new BigNumber(0);
+    let totalFeeDAI = new BigNumber(0);
+
+    try {
+        const signer = await getSigner(network);
+        const address = await signer.getAddress();
+        const isWalletAuthed = await getWalletAuthorizationStatus(network, address);
+        const isCollateralAuthed = await getCollateralAuthorizationStatus(network, auction.collateralType, address);
+
+        if (!isWalletAuthed) {
+            totalFeeETH = totalFeeETH.plus(fees.authTransactionFeeETH);
+            totalFeeDAI = totalFeeDAI.plus(fees.authTransactionFeeDAI);
+        }
+        if (!isCollateralAuthed) {
+            totalFeeETH = totalFeeETH.plus(fees.biddingTransactionFeeETH);
+            totalFeeDAI = totalFeeDAI.plus(fees.biddingTransactionFeeDAI);
+        }
+    } catch (e) {
```

Comment:
In which case it is cached?

Reply:
If no wallet is set. Because I execute `getSigner()`, which does not exist if you for example visit the frontend without connecting your wallet.

Reply:
> If no wallet is set

That's your expectation, but since it will be cached also on the network error or in any other case, this may be misleading?

Anyway, if no wallet is connected, there should be _two_ auth transactions included, one for the wallet and the other one for the collateral

Reply:
As discussed in call. Error handling is something we need to improve throughout the project at some point. However my implementation here for now is fine.